runtime-decompiler/src/main/java/org/jrd/backend/data/ArchiveManager.java

```diff
@@ -265,16 +265,19 @@ public void recursiveZip(File f2zip, String fName, ZipOutputStream zOut) throws
 
             return;
         }
-        FileInputStream fis = new FileInputStream(f2zip);
-        ZipEntry zEntry = new ZipEntry(fName);
-        zOut.putNextEntry(zEntry);
-        byte[] bytes = new byte[1024];
-        int length;
-        while ((length = fis.read(bytes)) >= 0) {
-            zOut.write(bytes, 0, length);
+
+        try (FileInputStream fis = new FileInputStream(f2zip);) {
+            ZipEntry zEntry = new ZipEntry(fName);
+            zOut.putNextEntry(zEntry);
+            byte[] bytes = new byte[1024];
+            int length;
+
+            while ((length = fis.read(bytes)) >= 0) {
+                zOut.write(bytes, 0, length);
+            }
+
+            zOut.closeEntry();
```

Comment:
my usuallparanoid, if the nested jars continue to work...

Reply:
@ilbinek what do you think?

Reply:
In theory this should not break anything. But with my experience with nested jars, I am afraid that theory is often not how it works out to be. I am not able to test it right now, so please do some basic tests - mainly editing. If those look ok, I am for this change. Unit tests testing this will hopefully be done in two weeks.

Reply:
So no body knows anything.. just try that ;)